Basic-Car-Maintenance/Shared/AppIntents/AddOdometerReadingAppIntent.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  AddOdometerReadingAppIntent.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Omar Hegazy on 23/08/2024.
+//
+
+import Foundation
+import AppIntents
+
+struct VehicleQuery: EntityQuery {
+   
+    func entities(for identifiers: [Vehicle.ID]) async throws -> [Vehicle] {
+        try await fetchVehicles()
+    }
+    
+    func suggestedEntities() async throws -> [Vehicle] {
+        try await fetchVehicles()
+    }
+    
+    private func fetchVehicles() async throws -> [Vehicle] {
+        let authViewModel = await AuthenticationViewModel()
+        let odometerVM = OdometerViewModel(userUID: authViewModel.user?.uid)
+        await odometerVM.getVehicles()
+        guard !odometerVM.vehicles.isEmpty else {
+            throw OdometerReadingError.emptyVehicles
+        }
+        return odometerVM.vehicles
+    }
+}
+
+enum DistanceUnit: String, AppEnum, CaseIterable {
+    static var typeDisplayRepresentation = TypeDisplayRepresentation(name: "Distance Type")
+    static var caseDisplayRepresentations: [DistanceUnit: DisplayRepresentation] {
+        [
+            .mile: "Miles",
+            .kilometer: "Kilometers"
+        ]
+    }
+    
+    case mile
+    case kilometer
+}
+
+struct AddOdometerReadingIntent: AppIntent {
```

Comment:
How does this work when it's not called anywhere? Is there a WWDC video I could reference to look at this?